controllers/drplacementcontrol_controller.go

```diff
@@ -1728,87 +1846,229 @@ func (r *DRPlacementControlReconciler) getOrCreateManagedClusterView(
 	return mcv, nil
 }
 
-func (d *DRPCInstance) listPVsFromS3Store(homeCluster string) ([]corev1.PersistentVolume, error) {
-	s3Bucket := constructBucketName(d.instance.Namespace, d.instance.Name)
-
-	return d.reconciler.PVDownloader.DownloadPVs(
-		d.ctx, d.reconciler.Client, d.reconciler.ObjStoreGetter, rmnutil.S3DownloadProfile(*d.drPolicy, homeCluster),
-		d.instance.Name, s3Bucket)
-}
-
-type ObjectStorePVDownloader struct{}
-
-func (s ObjectStorePVDownloader) DownloadPVs(ctx context.Context, r client.Reader,
-	objStoreGetter ObjectStoreGetter, s3Profile string,
-	callerTag string, s3Bucket string) ([]corev1.PersistentVolume, error) {
-	objectStore, err := objStoreGetter.objectStore(ctx, r, s3Profile, callerTag)
-	if err != nil {
-		return nil, fmt.Errorf("error when downloading PVs, err %w", err)
-	}
-
-	return objectStore.downloadPVs(s3Bucket)
-}
-
 func (d *DRPCInstance) advanceToNextDRState() {
-	var nextState rmn.DRState
+	lastDRState := d.getLastDRState()
+	nextState := lastDRState
 
-	switch d.instance.Status.LastKnownDRState {
-	case rmn.DRState(""):
-		nextState = rmn.Initial
+	switch lastDRState {
+	case rmn.Deploying:
+		nextState = rmn.Deployed
 	case rmn.FailingOver:
 		nextState = rmn.FailedOver
 	case rmn.FailingBack:
 		nextState = rmn.FailedBack
 	case rmn.Relocating:
 		nextState = rmn.Relocated
-	case rmn.Initial:
+	case rmn.Deployed:
 	case rmn.FailedOver:
 	case rmn.FailedBack:
 	case rmn.Relocated:
-	default:
-		nextState = rmn.DRState("")
 	}
 
 	d.setDRState(nextState)
 }
 
-func (d *DRPCInstance) resetDRState() {
-	d.log.Info("Resetting last known DR state", "lndrs", d.instance.Status.LastKnownDRState)
+func (d *DRPCInstance) setDRState(nextState rmn.DRState) {
+	if d.instance.Status.Phase != nextState {
+		d.log.Info(fmt.Sprintf("Phase: Current '%s'. Next '%s'",
+			d.instance.Status.Phase, nextState))
 
-	d.setDRState(rmn.DRState(""))
+		d.instance.Status.Phase = nextState
+		d.updateConditions()
+		d.needStatusUpdate = true
+	}
 }
 
-func (d *DRPCInstance) setDRState(nextState rmn.DRState) {
-	if d.instance.Status.LastKnownDRState != nextState {
-		d.log.Info(fmt.Sprintf("LastKnownDRState: Current State '%s'. Next State '%s'",
-			d.instance.Status.LastKnownDRState, nextState))
+func (d *DRPCInstance) updateConditions() {
+	d.log.Info(fmt.Sprintf("Current Conditions '%v'", d.instance.Status.Conditions))
```

Comment:
IIUC here we are relying mainly on DRPC.Status.Phase to update the conditions. One possible situation which I am trying to understand is this.

Ex: runInitialDeployment() to deploy VRG (and even the app) initially in a cluster.

- We move the state (i.e. DRPC.Status.Phase) to rmn.Deploying after running the placementRule and it making 
  a decision.
-  We create the VRG ManifestWork (say creation of the ManifestWork is success)
-  Upate the User placement rule (which is when OCM gets to know about the cluster where it needs to deploy the app)
- Then we advance to the next state of rmn.Deployed (assuming above operations to be successful)

However, lets say that even though creating ManifestWork for VRG was successful, later the managed cluster failed to successfully create the VRG (or say VRG conditions says its Available.Status = False due to some problems while reconciling). 

In such situations from DRPC perspective its Status.Phase is "Deployed" and even its conditions are updated based on DRPC.Status.Phase. VRG conditions are obtained while updating the DRPC status and displayed.

So, my question is, should DRPC.Status.Phase or/and DRPC.Status.Conditions content be based on VRG conditions? i.e. should they be more tightly coupled with VRG conditions instead of DRPC.Status just displaying the VRG conditions?

Let me know if any of my understanding is not correct or if I have missed something.

It does not look urgent. But curious to know some perspectives on this.

Reply:
VRG conditions may change as PVCs are added, also initial state of empty VRG creation will report success (and is monitored). The real status update in DRPC for VRG would be once we refresh it at sync intervals, where we fetch the VRG state, including last sync time for volumes, and reflect that in DRPC state.

From the perspective of failover or relocate, we check secondary status to reach desired state, so that helps.

Ultimately a "degraded" condition to reflect a degraded state for DRPC when VRG is degraded can help or be added subsequently.

Reply:
ACK